### DISCUSSION TITLE
How to debug in the internal Terminal instead of in the Debug Console?

### DISCUSSION THREAD

**Discussion by EXStevens** [microsoft/vscode-cpptools#8605] [Q&A] [answered]
The console/terminal settings in launch.json seem that they're no effect.

**Comment by TryerGit** [comment, marked as answer]
AFAIK, the output from your program does **NOT** appear in the integrated terminal under Windows. It appears in the debug console. The problem with the debug console is that it does not accept user input, scanf, getchar, etc. See https://stackoverflow.com/questions/64394561/how-to-type-into-the-integrated-terminal-on-vs-code-in-windows
So, the only solution in Windows is to have the output appear on a separate console screen, much like how the output appears there in Visual Studio IDE, for instance. Here, you can enter your user input.

In Linux, the output can appear in a new session of the integrated terminal where you can enter user input.

**Reply by EXStevens** [reply]
Thanks.
I used to use it on Linux so...;
